bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/internal/RingHandlerFactory.java

```diff
@@ -0,0 +1,130 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ring.internal;
+
+import static org.openhab.binding.ring.RingBindingConstants.*;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ring.handler.AccountHandler;
+import org.openhab.binding.ring.handler.ChimeHandler;
+import org.openhab.binding.ring.handler.DoorbellHandler;
+import org.openhab.binding.ring.handler.StickupcamHandler;
+import org.openhab.core.net.HttpServiceUtil;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RingHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Wim Vissers - Initial contribution
+ * @author Chris Milbert - Stickupcam contribution
+ */
+
+@Component(service = { ThingHandlerFactory.class,
+        RingHandlerFactory.class }, immediate = true, configurationPid = "binding.ring")
+public class RingHandlerFactory extends BaseThingHandlerFactory {
+    private Logger logger = LoggerFactory.getLogger(RingHandlerFactory.class);
+    private final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS;
+
+    private NetworkAddressService networkAddressService;
+
+    @Nullable
+    private HttpService httpService;
+    private int httpPort;
+    private ComponentContext componentContext;
+
+    @Activate
+    public RingHandlerFactory(@Reference NetworkAddressService networkAddressService,
+            @Reference HttpService httpService, ComponentContext componentContext) {
+        super.activate(componentContext);
+        httpPort = HttpServiceUtil.getHttpServicePort(componentContext.getBundleContext());
+        if (httpPort == -1) {
+            httpPort = 8080;
+        }
+        this.httpService = httpService;
+        this.networkAddressService = networkAddressService;
+
+        logger.debug("Using OH HTTP port {}", httpPort);
+
+        SUPPORTED_THING_TYPES_UIDS = new HashSet<>();
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_ACCOUNT);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_DOORBELL);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_CHIME);
+        SUPPORTED_THING_TYPES_UIDS.add(THING_TYPE_STICKUPCAM);
```

Comment:
`SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_ACCOUNT, THING_TYPE_DOORBELL.....);`